src/controllers/agent.js

```diff
@@ -0,0 +1,54 @@
+import Model from '../models/model';
+const jwt = require('jsonwebtoken');
+const bcrypt = require('bcrypt');
+import dotenv from 'dotenv'
+const agentModel = new Model('agents');
+dotenv.config();
+
+export const signupAgent = async (req, res) => {
+  const { firstName, lastName, password, email, phoneNumber } = req.body;
+  const columns = 'first_name, last_name, password, email, phone_number';
+  const strong = await bcrypt.hash(password, 12);
+  const values = `'${firstName}', '${lastName}', '${strong}', '${email}', '${phoneNumber}'`;
+  try {
+  const checkEmail = await agentModel.select(
+    '*',
+    ` where email = '${email}' `
+  );
+    if (checkEmail.rowCount > 0 ) {
+      return res.status(409).json({ message: 'Account already Exist' });
+    }
+    const data = await agentModel.insertWithReturn(columns, values);
+    const {id} = data.rows[0]
+    const newUser = { id, firstName, lastName, email};
+
+  const token = jwt.sign({ newUser}, process.env.TOKEN_KEY, {
```

Comment:
This logi to sign JWT should be in a helper function.

Reply:
Resolved already moved too helper function.